src/struct/hash.h

```diff
@@ -15,16 +15,17 @@
  */
 
 #include <stdint.h>
+#include <asm/bitsperlong.h>
```

Comment:
This header is Linux-specific and not portable.

Reply:
I thought so too until I saw [this](https://github.com/taviso/ctypes.sh/blob/091e64421507c3502b38b1b88d78b12b4269e80e/lib/asm/bitsperlong.h)... :)

Reply:
Aha.  Well, I'd still suggest `__LP64__` and `__ILP32__`.  Unlike the stub asm/bitsperlong.h, they work with 32- and 64-bit architectures other than x86.  Tavis might have a different opinion.

Reply:
I'll take a shot at changing this to use __ILP32__ and __LP64__ and cleaning this up throughout.  Thanks. :)